atomics/include/desul/atomics/Compare_Exchange_OpenMP.hpp

```diff
@@ -0,0 +1,101 @@
+/* 
+Copyright (c) 2019, Lawrence Livermore National Security, LLC
+and DESUL project contributors. See the COPYRIGHT file for details.
+Source: https://github.com/desul/desul
+
+SPDX-License-Identifier: (BSD-3-Clause)
+*/
+#ifndef DESUL_ATOMICS_COMPARE_EXCHANGE_OPENMP_HPP_
+#define DESUL_ATOMICS_COMPARE_EXCHANGE_OPENMP_HPP_
+#include "desul/atomics/Common.hpp"
+#include<cstdio>
+
+#ifdef DESUL_HAVE_OPENMP_ATOMICS
+namespace desul {
+
+#if _OPENMP > 201800
+// atomic_thread_fence for Core Scope
+inline void atomic_thread_fence(MemoryOrderSeqCst, MemoryScopeCore) {
+  #pragma omp flush acq_rel
```

Comment:
You should comment that this is not a typo.

Reply:
It isn't? Why isn't `seq_cst` desired in the `seq_cst` fence?

Reply:
I see the comment now, but `seq_cst` has been available since well before 4.0, and `acq_rel` came in after that in 5.0.  Is there a compiler that supports `acq_rel` but not `seq_cst`?

Reply:
And I'm wrong, OpenMP's flush doesn't have seq_cst, only atomic does.  The `seq_cst` behavior of flush is referred to as a strong flush, and is spelled `#pragma omp flush` with no memory order.  I'll follow up with some folks on why we don't let the user spell it the specific way, but that's where we are.

Reply:
ok changed that.